packages/react-form-state/src/FormState.ts

```diff
@@ -186,52 +180,103 @@ export default class FormState<
   }
 
   @memoize()
-  private getHandlersForField(path: keyof FieldStates<Fields>) {
-    return {
-      onChange: this.updateField.bind(this, path),
-      onBlur: this.blurField.bind(this, path),
-    };
+  @bind()
+  private fieldWithHandlers<Key extends keyof Fields>(
+    field: FieldStates<Fields>[Key],
+    fieldPath: Key,
+  ) {
+    return Object.assign({}, field, {
```

Comment:
Can you use the spread operator here?

Reply:
Agreed, even if you need to cast it is better to use the feature you want to use (also, apparently spread will be faster than object assign somehow /shrug)

Reply:
I avoided spread in cases where I would need to cast to satisfy typescript.

Reply:
Yeah I had the same issues with the spread operator and casting

Reply:
I guess we can just do it with spread, I just really don't like having casts all over the place as it feels messy and error prone.